server/rooms.ts

```diff
@@ -761,6 +768,13 @@ export abstract class BasicRoom {
 			this.reportJoin('j', user.getIdentityWithStatus(this.roomid), user);
 		}
 
+		if (user.can('mute', null, this)) {
+			this.sendUser(
+				user,
+				this.getStaffIntroMessage(user)
+			);
+		}
```

Comment:
```suggestion
		const staffIntro = this.getStaffIntroMessage(user);
		if (staffIntro) this.sendUser(user, staffIntro);
```

(And the same for `onRename`)

Is `onJoin` the right place for this, though? It seems wrong to me.

Reply:
I'm not entirely sure. It was originally in `onConnect` which made certain things not appear on autojoin since the user wasn't logged in yet. I moved it to `onJoin` because it sounded correct and ended up working when I tried it there... where do you suggest it go?

Reply:
`onConnect` happens when you open the room in any connection. `onJoin` happens whenever the "user joined" message appears - when you go from a guest to logged in, or when you join a room while logged in (so only the _first_ time you open the room).